dask/highlevelgraph.py

```diff
@@ -499,6 +499,14 @@ def _repr_html_(self, layer_index="", highlevelgraph_key=""):
         else:
             shortname = self.__class__.__name__
 
+        from dask.array.svg import svg
```

Comment:
Very likely we want this import to live somewhere else before merging (I know you had some problems with the imports earlier)

Reply:
I tried several things. No elegant solution was found.

Turns out, it's a case of Circular Imports.
Trying to fix it rn. 

`dask.array` imports `dask.delayed` which imports `HighLevelGraph` which in turn imports `dask.array`

```
ImportError                               Traceback (most recent call last)
<ipython-input-1-dc46e0049dfc> in <module>
----> 1 import dask.array as da
      2 
      3 x = da.random.random((10000, 10000), chunks=(100, 100))
      4 y = x + x.T - x.mean(axis=0)
      5 y.dask

~/GitHub/dask/dask/__init__.py in <module>
      3 from .base import annotate, compute, is_dask_collection, optimize, persist, visualize
      4 from .core import istask
----> 5 from .delayed import delayed
      6 from .local import get_sync as get
      7 

~/GitHub/dask/dask/delayed.py in <module>
     18 from .context import globalmethod
     19 from .core import quote
---> 20 from .highlevelgraph import HighLevelGraph
     21 from .optimization import cull
     22 from .utils import OperatorMethodMixin, apply, ensure_dict, funcname, methodcaller

~/GitHub/dask/dask/highlevelgraph.py in <module>
     23 from .utils import _deprecated, ensure_dict, key_split, stringify
     24 from .utils_test import add, inc  # noqa: F401
---> 25 from dask.array.svg import svg
```

I believe, that's why importing `svg` inside the method works for now as it has no conflicts.

Reply:
Potentially we could move the svg.py file up, so it doesn't live under the array module anymore. Or keep this solution, delaying the import until inside the function (although I think it should probably have the relative import path with dots, instead of starting with dask dot ...)

We should think about this some more, and then try and work out what the cleanest solution is.

Reply:
In the future I would like to use classes for this.  I would expect array layers to be of some type, like `ArrayLayer`, and then I would expect to call `layer.get_fancy_repr()` or something similar and then the array layers can call the array code dataframe layers can call the dataframe code, etc..

This isn't built out yet though, so we'll probably have to be sloppy until then.

Reply:
Just for testing, I moved `svg.py` from `dask/array/svg.py` to `dask/svg.py` and then imported that using
```py
from .svg import svg
```
and it works (on top) :tada:. If we plan on using this fix, I will have to change a little code in some files too to make sure the right file is getting imported!

------

Also, I have edited the current absolute addressing to relative and it still works!
```py
from .array.svg import svg
```
works **inside the method**.

Reply:
If we go with the import happening inside the method here, then I think it makes most sense to put it in this block:

```python
            if chunks is not None:
                from .array.svg import svg
                svg_repr = "<br />" + svg(chunks)
```

Reply:
1. Moved the import inside, so it only gets called when needed.
2. Surpringly, it's taking the same time as the current Dask implementation.

|                                   dask/dask                                   |                                  freyam/dask                                  |
|:-----------------------------------------------------------------------------:|:-----------------------------------------------------------------------------:|
| 33.7 ns ± 0.743 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each) | 34.3 ns ± 0.615 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each) |


Calculated locally using `%timeit` on the following code.

```py
import dask.array as da

XL = da.ones((67552, 365941), chunks=(652, 5792))
XL2 = XL.T.astype("f4")
XC = da.ones((365941, 26), chunks=(365941, 26))
LS = da.linalg.lstsq(XC, XL2)[0]
XC_LS = XC @ LS
XLP = XL2 - XC_LS

XLP.dask
```

Reply:
You'll want timeit only on the very last line `XLP.dask`

Reply:
Yes, I had done that.